source/exe/hot_restart.h

```diff
@@ -201,5 +122,39 @@ class HotRestartImpl : public HotRestart,
   bool parent_terminated_{};
 };
 
+#else  // __linux__
+
+/**
+ * No-op implementation of HotRestart for everybody else.
+ */
+class HotRestartImpl : public HotRestart, public Stats::RawStatDataAllocator {
+public:
+  HotRestartImpl(Options& options);
+
+  Thread::BasicLockable& logLock() { return log_lock_; }
+  Thread::BasicLockable& accessLogLock() { return access_log_lock_; }
+
+  // Server::HotRestart
+  void drainParentListeners() override {}
+  int duplicateParentListenSocket(const std::string&) override { return -1; }
+  void getParentStats(GetParentStatsInfo& info) override { memset(&info, 0, sizeof(info)); }
+  void initialize(Event::Dispatcher&, Server::Instance&) override {}
+  void shutdownParentAdmin(ShutdownParentAdminInfo&) override {}
+  void terminateParent() override {}
+  void shutdown() override {}
+  std::string version() override { return "disabled"; }
+
+  // RawStatDataAllocator
+  Stats::RawStatData* alloc(const std::string& name) override;
+  void free(Stats::RawStatData& data) override;
+
+private:
+  SharedMemory& shmem_;
```

Comment:
IMO the NOP hot restart impl should not use shared memory or the process shared mutex stuff. Basically, I would just do exactly what https://github.com/lyft/envoy/blob/master/test/integration/server.cc#L93 does which is to use process local locks, the heap allocator, etc. I think you can probably revert all the changes related to shared memory and just move the test impl into prod code somewhere and use that.